src/cascade/dismod/db/wrapper.py

```diff
@@ -187,15 +181,40 @@ def __dir__(self):
         attributes.extend(super().__dir__())
         return attributes
 
+    def update_table_columns(self, table_definition, table):
+        new_columns = table.columns.difference(table_definition.c.keys())
+        new_columns = {c: table.dtypes[c] for c in new_columns}
+
+        bad_column_names = [c for c in new_columns.keys() if not c.startswith("x_")]
+        if bad_column_names:
+            raise ValueError("Covariate column names must start with 'x_'")
+
+        if table_definition.name == "avgint":
+            add_columns_to_avgint_table(self._metadata, new_columns)
+        elif table_definition.name == "data":
+            add_columns_to_data_table(self._metadata, new_columns)
+        else:
+            raise ValueError(f"Can't add columns to {table.name}")
+
     def __getattr__(self, table_name):
         if table_name in self._table_data:
             return self._table_data[table_name]
         elif table_name in self._table_definitions:
             if self.engine is None:
                 raise ValueError("Cannot read from disk before an engine is set")
             table = self._table_definitions[table_name]
-            with self.engine.connect() as conn:
-                data = pd.read_sql_table(table.name, conn)
+            try:
+                with self.engine.connect() as conn:
+                    data = pd.read_sql_table(table.name, conn)
+            except ValueError as e:
+                if str(e) != f"Table {table.name} not found":
+                    raise
+                data = pd.DataFrame({k: pd.Series(dtype=v.type.python_type) for k, v in table.c.items()})
```

Comment:
There is a function in this class, called `empty_table` that doesn't have the same list of column types, and it should.

Reply:
Good catch, I totally forgot about `empty_table`. This should go through that path.